content/en/blog/_posts/2024-09-25-swap-ga-graduation/2024-09-25-swap-ga-graduation.md

```diff
@@ -0,0 +1,445 @@
+```
+
+The currently available configuration options for `swapBehavior` are:
+- `NoSwap` (default): Kubernetes workloads cannot use swap. However, processes
+  outside of Kubernetes' scope, like system daemons (such as kubelet itself!) can utilize swap.
+  This behavior is beneficial for protecting the node from system-level memory spikes,
+  but it does not safeguard the workloads themselves from such spikes.
+- `LimitedSwap`: Kubernetes workloads can utilize swap memory, but with certain limitations.
+  The amount of swap available to a Pod is determined automatically,
+  based on the proportion of the memory requested relative to the node's total memory.
+  Only non-high-priority Pods under the [Burstable](/docs/concepts/workloads/pods/pod-qos/#burstable)
+  Quality of Service (QoS) tier are permitted to use swap.
+  For more details, see the [section below](#how-is-the-swap-limit-being-determined-with-limitedswap).
+
+If configuration for `memorySwap` is not specified,
+by default the kubelet will apply the same behaviour as the `NoSwap` setting.
+
+On Linux nodes, Kubernetes only supports running with swap enabled for hosts that use cgroup v2.
+On cgroup v1 systems, all Kubernetes workloads are not allowed to use swap memory.
+
+## Install a swap-enabled cluster with kubeadm
+
+### Before you begin
+
+It is required for this demo that the kubeadm tool be installed, following the steps outlined in the
+[kubeadm installation guide](/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm).
+If swap is already enabled on the node, cluster creation may proceed.
+If swap is not enabled, please refer to the provided instructions for enabling swap.
+
+### Create a swap file and turn swap on
+
+I'll demonstrate creating 4GiB of swap, both in the encrypted and unencrypted case.
+
+#### Setting up unencrypted swap
+
+An unencrypted swap file can be set up as follows.
+
+```bash
+dd if=/dev/zero of=/swapfile bs=128M count=32
+chmod 600 /swapfile
+mkswap /swapfile
+swapon /swapfile
+```
+
+#### Setting up encrypted swap
+
+An encrypted swap file can be set up as follows.
+Bear in mind that this example uses the `cryptsetup` binary (which is available
+on most Linux distributions).
+
+```bash
+dd if=/dev/zero of=/swapfile bs=128M count=32
+chmod 600 /swapfile
+cryptsetup --type plain --cipher aes-xts-plain64 --key-size 256 -d /dev/urandom open /swapfile cryptswap
+mkswap /dev/mapper/cryptswap
+swapon /dev/mapper/cryptswap
+```
+
+#### Verify that swap is enabled
+
+Swap can be verified to be enabled with both `swapon -s` command or the `free` command
+
+```shell
+> swapon -s
+Filename				Type		Size		Used		Priority
+/dev/dm-0                               partition	4194300		0		-2
+```
+
+```shell
+> free -h
+               total        used        free      shared  buff/cache   available
+Mem:           3.8Gi       1.3Gi       249Mi        25Mi       2.5Gi       2.5Gi
+Swap:          4.0Gi          0B       4.0Gi
```

Comment:
We should separate the commands from output; everything marked as shell should be valid POSIX shell (or close).